bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,23 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r"""
+            (https?://                  # http:// or https:// as only these
+                                        # 2 are supported by the ``requests``
+                                        # library
+            [^.:%\s_/?#[\]@\\]+         # Initial Part of Domain
+            \.                          # A required dot (.)
+            (
+                (?:[^\s()%\'"`<>|\\]+)  # Path Name
+            |                           # OR
+                \([^\s()%\'"`<>|\\]*\)  # Path Name contained within ()
```

Comment:
This is not an acceptable comment yet - it only explains what is extremely obvious, which are not appropriate comments.
We need the comments to explain the complex aspects.

why contained within () ?

My understanding is that that both parts of this `OR` is to prevent matching a trailing `)` after the URL, such as `(e.g. http://foo.bar/path)`

This part of the regex ensures that it only matches balanced round brackets within the URL. If the URL actually should contain unbalanced round brackets, this regex will skip the URL, because it cant reliably find the end of the URL.
This part of the regex is sub-optimal, but is 'good enough' and doesnt have false positives.